x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/features/downsample.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { setupEnvironment } from '../../helpers';
+import { DownsampleTestBed, setupDownsampleTestBed } from './downsample.helpers';
+
+describe('<EditPolicy /> downsample', () => {
+  let testBed: DownsampleTestBed;
+  const { httpSetup, httpRequestsMockHelpers } = setupEnvironment();
+
+  beforeEach(async () => {
+    httpRequestsMockHelpers.setDefaultResponses();
+
+    await act(async () => {
+      testBed = await setupDownsampleTestBed(httpSetup);
+    });
+
+    const { component } = testBed;
+    component.update();
+  });
+
+  test('enabling downsample in warm should hide readonly in warm and cold', async () => {
+    const { actions } = testBed;
+
+    await actions.togglePhase('warm');
+    await actions.togglePhase('cold');
+
+    expect(actions.warm.downsample.exists()).toBeTruthy();
+    expect(actions.warm.readonlyExists()).toBeTruthy();
+    expect(actions.cold.downsample.exists()).toBeTruthy();
+    expect(actions.cold.readonlyExists()).toBeTruthy();
+
+    await actions.warm.downsample.toggle();
+
+    expect(actions.warm.readonlyExists()).toBeFalsy();
+    expect(actions.cold.readonlyExists()).toBeFalsy();
+  });
+
+  test('enabling downsample in hot should hide readonly in hot', async () => {
```

Comment:
It looks like `readonly` is also disabled for `warm`, and `cold` in this scenario. Can you update the test if that's correct?